src/main/java/io/confluent/kafkarest/resources/TopicsResource.java

```diff
@@ -74,6 +58,37 @@ public Topic getTopic(@PathParam("topic") String topicName) {
     return topic;
   }
 
+  @GET
+  @Path("/{topic}/partition/{partition}/messages")
+  @PerformanceMetric("topic.consume-binary")
+  @Produces({Versions.KAFKA_V1_JSON_BINARY_WEIGHTED,
+             Versions.KAFKA_V1_JSON_WEIGHTED,
+             Versions.KAFKA_DEFAULT_JSON_WEIGHTED,
+             Versions.JSON_WEIGHTED,
+             Versions.KAFKA_V1_JSON_BINARY_WEIGHTED_LOW,
+             Versions.ANYTHING})
+  public void consumeBinary(final @Suspended AsyncResponse asyncResponse,
+                            final @PathParam("topic") String topicName,
+                            final @PathParam("partition") int partitionId,
+                            final @QueryParam("offset") long offset,
+                            final @QueryParam("count") @DefaultValue("1") long count) {
```

Comment:
Well, it was precisely for debugging purposes that I needed this API, that's why I defaulted to 1. But I can understand why it wouldn't be useful for other purposes.

Before defaulting to 1, I simply made the API return all the messages contained in the Kafka response (i.e. within ConsumerConfig.fetchMessageMaxBytes bytes). Maybe it would be the best behavior ?